### PR TITLE
docs/tests: feature_intervention_generate logits changed from 3D to 2D - document / test this

### PR DESCRIPTION
**Pull request by hijohnnylin**
When adding support for nnsight engine, we slightly changed the return shape of feature_intervention_generate's logits tensor. It changed from batch, seq_len, vocab_size to just seq_len, vocab_size. This broke downstream code/users that assumed the previous shape.

Previous: No squeeze(0), returned logits is 2D
https://github.com/decoderesearch/circuit-tracer/blob/e49c213cf5238af7543b7d433d5ed5ca7b597e79/circuit_tracer/replacement_model.py#L900

New: squeeze(0). returned logits is 3D
https://github.com/decoderesearch/circuit-tracer/blob/9317b2aaca533dad6cdef9b206b8acc3d542eb5b/circuit_tracer/replacement_model/replacement_model_transformerlens.py#L900

The new shape is fine since the batch size is always 1 anyway, so this commit is a Claude-assisted update that doesn't update the behavior - it only adds documentation about this as well as a few tests and new asserts.
Possibly it's too verbose a PR for the relatively small change, so feel free to edit/change/minimize it.

(Downstream we added a check to ensure it's dimension 2 which is our workaround for now.)

### CODE REVIEW

**Review by hannamw — APPROVED** [review]
Thanks for documenting this change! I've fixed the pyright issue, so this should be good to go now
